src/ParallelAlgorithms/EventsAndTriggers/EventsAndTriggers.hpp

```diff
@@ -40,12 +50,24 @@ class EventsAndTriggers {
                   Parallel::GlobalCache<Metavariables>& cache,
                   const ArrayIndex& array_index,
                   const Component* component) const noexcept {
+    using compute_tags = tmpl::remove_duplicates<tmpl::filter<
+        tmpl::flatten<tmpl::transform<
+            tmpl::at<typename Metavariables::factory_creation::factory_classes,
+                     Event>,
+            get_tags<tmpl::_1>>>,
+        db::is_compute_tag<tmpl::_1>>>;
+    std::optional<ObservationBox<db::detail::expand_subitems<compute_tags>,
```

Comment:
Perhaps get the type with `decltype()`? We probably shouldn't use `db::detail` here..

Reply:
sorry, I'm not sure how `decltype()` is supposed to replace `expand_subitems`. Could you clarify? I can also just move `expand_subitems` out of the `detail` namespace :shrug:

Reply:
I was thinking maybe something like `decltype(make_observation_box<compute_tags>(declval<DataBoxType>()))` could work

Reply:
I like @nilsleiffischer's suggestion, but moving `expand_subitems` is also fine with me.